### DISCUSSION TITLE
Create an R package where the main function launches a targets pipeline

### DISCUSSION THREAD

**Discussion by gorkang** [ropensci/targets#548] [Help]
I want to create an R package from a [project](https://github.com/gorkang/jsPsychMonkeys) where the main functionality is launching a targets pipeline (to simulate participants in a jsPsych experiment). 

Right now I use it as a standalone R Studio project. My present workflow is: 

1. Open the project
2. Change the necessary parameters in a "configuration target" on `_targets.R`
3. `targets::tar_make()`

The first target of the pipeline is a "configuration" function where all the relevant parameters for all the targets are set (for example, the local folder where the jsPsych protocol is and the number of participants to run). 

I would like to encapsulate everything in an R package. Maybe the "configuration" function could be the main function in the package. And when run, it should start the target pipeline. For example:

`jsPsychMonkeys::simulate_participants(local_folder  = "experiment1", n_participants = 100)`

Is something like this possible, or am I way over my head?

Thanks!

**Comment by wlandau** [comment]
What you describe sounds possible, but I am not sure what the goal is. If the goal is to make an existing analysis portable, then it would probably be easier to share it in a Git/GitHub repo with an `renv` library and directions to run the analysis. If you plan to share it with other developers, the [Target Markdown](https://books.ropensci.org/targets/markdown.html) will help them understand the analysis. If you want to establish a pattern of similar analyses, then you can write a package with target factories to make it easier to create pipelines. https://wlandau.github.io/targetopia/contributing.html has general advice on this, and https://docs.ropensci.org/stantargets/ is an example package. I have found target factories to be a more successful encapsulation technique than wrappers around `tar_make()`.

**Comment by gorkang** [comment]
Thanks for your answer @wlandau 

The way targets forces you to work, and some of its tools (e.g. skipping nodes when nothing changes, `tar_visnetwork()`, etc.) are so useful for complex projects, I started using it for **non-analysis** projects like this one. 

In this case, it is a system to simulate participants using docker and Rselenium. The functionality is general enough that I want to make it easier for me and others to run synthetic participants (i.e. simulate data. This is crucial to test data preparation and analysis scripts before any human completes the experiments).

Given we have multiple experimental protocols, each one with its Rstudio project for data preparation and analysis, I need to simulate data for a bunch of protocols, and I would love to do this using something like `jsPsychMonkeys::simulate_participants(local_folder = "experiment1", n_participants = 100)` somewhere inside the "data analysis" project, instead of having to do it in a separate "simulate data" project.

Does this make sense?

**Reply by wlandau** [reply]
I do not think I have the domain expertise to fully understand the use case, but nothing specific stands out to me as unwise. If it appears to work, for it.
